CMakeLists.txt

```diff
@@ -13,6 +13,14 @@ option(ONNX_MLIR_SUPPRESS_THIRD_PARTY_WARNINGS "Suppress warning in third_party
 
 set(CMAKE_CXX_STANDARD 17)
 
+# New in cmake 3.15:
+# https://cmake.org/cmake/help/latest/policy/CMP0094.html
+# Stops python lookup as soon as a version satisfying version constraints
+# is found.
+if (POLICY CMP0094)
+  cmake_policy(SET CMP0094 NEW)
+endif()
```

Comment:
If this requires 3.15, the minimum required cmake version (currently 3.13.4) should be bumped up to 3.15.

Reply:
@gongsu832 would it be fine to bump to 3.15? Or are we help back to lower version because of some dependence?

@nhat-nguyen would you be willing to try moving us up to 3.15 if appropriate? Tx

Reply:
definitely, i can try moving us up to 3.15

Reply:
Actually we currently use 3.13.4 because that's what llvm uses. There is a comment in llvm that starting from llvm 17 they will require cmake 20.2 and they suggested to move up to 20.2 directly. This would require we move our Ubuntu base docker image from focal to jammy. I will create a PR to do that.

Reply:
@gongsu832 please link the PR so @nhat-nguyen will know when we may proceed with this one, thanks